internal/auth/auth.go

```diff
@@ -37,3 +40,127 @@ func AuthHandler(c *gin.Context) {
 		RequestID: requestId,
 	})
 }
+
+func RegisterUser(c *gin.Context) {
+	var request u.RegisterUserRequest
+
+	if err := c.ShouldBindJSON(&request); err != nil {
+		c.JSON(http.StatusBadRequest, u.RegisterUserResponse{
+			Status: *models.StatusFailed(fmt.Sprintln("Invalid inputs")),
+		})
+		return
+	}
+
+	registerUser := &u.CheckIfUserExistsRequest{
+		EmailId: request.EmailId,
+	}
+
+	userRepo := env.FromContext(c).UserRepository
+	user, err := userRepo.CheckIfUserExists(registerUser)
+
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, u.RegisterUserResponse{
+			Status: *models.StatusSomethingWentWrong(),
+		})
+		return
+	}
+
+	if user.IsExisting {
+		c.JSON(http.StatusConflict, u.RegisterUserResponse{
+			Status:    *models.StatusSuccess(),
+			IsNewUser: user.IsExisting,
+		})
+		return
+	}
+
+	hashedPassword, err := util.HashPassword(request.Password)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, u.RegisterUserResponse{
+			Status: *models.StatusSomethingWentWrong(),
+		})
+		return
+	}
+
+	firstName, lastName := util.GetFirstNameAndLastName(request.FullName)
+	newUserData := &models.User{
+		FirstName:    firstName,
+		LastName:     lastName,
+		EmailId:      request.EmailId,
+		PasswordHash: hashedPassword,
+	}
```

Comment:
I think Gunjan Sir mentioned last time that the user will input first and last names in separate fields